atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -172,15 +195,15 @@ private long getBackoffTimeWhenSweepHasNotRun() {
     }
 
     @VisibleForTesting
-    boolean runOnce() {
+    SweepOutcome runOnce() {
         Optional<TableToSweep> tableToSweep = getTableToSweep();
         if (!tableToSweep.isPresent()) {
             // Don't change this log statement. It's parsed by test automation code.
             log.debug("Skipping sweep because no table has enough new writes to be worth sweeping at the moment.");
-            return false;
+            return SweepOutcome.NOTHING_TO_SWEEP;
         } else {
             specificTableSweeper.runOnceAndSaveResults(tableToSweep.get());
```

Comment:
This can throw a runtimeException, we should catch it and set OutCome as `ERROR`.

Reply:
Just realised the runtimeexception is caught above. It might be better to catch it here, but its fine in the caller too.